### PR TITLE
Install HTCondor autoscaler into filesystem and fix node deletion bug

### PR DESCRIPTION
**Pull request by tpdownes**
- Enhance htcondor-configure
  - anticipate that access point (not central manager) will run autoscaler in most installations
  - there are some harmless variable name changes to harmonize style
- Enhance htcondor-install
  - fix bug in node deletion from regional MIGs (status quo simply doesn't work)
  - create runner that installs a Python virtual environment with autoscaler dependencies
  - create runner that installs the actual autoscaler Python code

This PR does not actually cause the autoscaler to execute. That will be in a forthcoming PR.

### Submission Checklist

* [X] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [X] Are all tests passing? (`make tests`)
* [ ] Have you written unit tests to cover this change?
* [X] Is unit test coverage still above 80%?
* [X] Have you updated all applicable documentation?
* [X] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

File: community/modules/scripts/htcondor-install/files/autoscaler.py [hunk 1]
```diff
@@ -295,22 +296,22 @@ def scale(self):
 
                     if server not in node_busy:
                         if status == "Unclaimed":
-                            node_busy[server] = False
+                            node_busy[server] = zone
```
Comment (heyealex) [inline community/modules/scripts/htcondor-install/files/autoscaler.py:299]: The `node_busy` map is not well named anymore. You could consider renaming it to something like `server_zone` or setting it to a tuple or map of (busy:bool, zone:string).

File: community/modules/scripts/htcondor-install/files/install-htcondor-autoscaler-deps.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+- name: Install but do not activate HTCondor autoscaler
+  become: true
+  hosts: localhost
+  tasks:
+  - name: Install Python 3 pip
+    ansible.builtin.yum:
```
Comment (heyealex) [inline community/modules/scripts/htcondor-install/files/install-htcondor-autoscaler-deps.yml:21]: Will this always be run on centos/redhat images? If not, ansible.buildtin.package would be a more flexible choice here.

**Review by heyealex — CHANGES_REQUESTED** [review]

File: community/modules/scheduler/htcondor-configure/variables.tf [hunk 3]
```diff
@@ -28,6 +28,7 @@ variable "access_point_roles" {
   description = "Project-wide roles for HTCondor Access Point service account"
   type        = list(string)
   default = [
+    "roles/compute.instanceAdmin",
```
[the remark this answers is not in the document]
Comment (tpdownes) [reply community/modules/scheduler/htcondor-configure/variables.tf:31]: In Slurm-speak: the central manager is the controller while the access point is a login node from which one can submit jobs. It is definitely not ideal to put this on the access point but the autoscaler.py presently requires it. Changing that is relatively minor fix but something for post-release.

Recall that, right now, POSIX users other than `root` and`condor` are blocked from the instance metadata server anyhow.

File: community/modules/scripts/htcondor-install/files/autoscaler.py [hunk 4]
```diff
@@ -87,12 +87,12 @@ def __init__(self, multizone=False):
...
 
     # Remove specified instance from MIG and decrease MIG size
-    def deleteFromMig(self, instance):
+    def deleteFromMig(self, instance, zone):
         instanceUrl = "https://www.googleapis.com/compute/v1/projects/" + self.project
-        if self.multizone:
-            instanceUrl += "/regions/" + self.region
-        else:
-            instanceUrl += "/zones/" + self.zone
+        # if self.multizone:
```
[the remark this answers is not in the document]
Comment (tpdownes) [reply community/modules/scripts/htcondor-install/files/autoscaler.py:92]: Will do.

File: community/modules/scripts/htcondor-install/files/install-htcondor-autoscaler-deps.yml (hunk 2 above)
Comment (tpdownes) [reply community/modules/scripts/htcondor-install/files/install-htcondor-autoscaler-deps.yml:21]: I will check the package names on Debian derivatives. If so, I will make the change. There are all sorts of CentOS 7 assumptions here and throughout but we should not be adding more.

Comment (tpdownes) [reply community/modules/scripts/htcondor-install/files/install-htcondor-autoscaler-deps.yml]: Resolved on HEAD of PR

File: community/modules/scripts/htcondor-install/files/autoscaler.py (hunk 1 above)
Comment (tpdownes) [reply community/modules/scripts/htcondor-install/files/autoscaler.py]: Resolved on HEAD of PR. Called it `idle_node_zones`. This code is in serious need of re-implementation (shell call-outs are unnecessary as HTCondor has Python interface).

File: community/modules/scripts/htcondor-install/files/autoscaler.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (tpdownes) [reply community/modules/scripts/htcondor-install/files/autoscaler.py]: Resolved on HEAD of PR

**Pull request merged**